### PR TITLE
cleanup and tests added in hibernate

### PR DESCRIPTION
**Pull request by mustafaiman**
Removes unused code in initialisation of Hibernate module / Cleanup in initialisation
Adds new tests

### CODE REVIEW

File: hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/instance/HazelcastAccessor.java [hunk 1]
```diff
@@ -69,10 +69,6 @@ public static HazelcastInstance getHazelcastInstance(final SessionFactory sessio
     public static HazelcastInstance getHazelcastInstance(final SessionFactoryImplementor sessionFactory) {
         final Settings settings = sessionFactory.getSettings();
         final RegionFactory rf = settings.getRegionFactory();
-        if (rf == null) {
-            LOGGER.severe("Hibernate 2nd level cache has not been enabled!");
-            return null;
-        }
```
Comment (mesutcelik) [inline hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/instance/HazelcastAccessor.java]: is `HazelcastAccessor` a test code? I see this is only used by test classes?

Comment (mustafaiman) [inline hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/instance/HazelcastAccessor.java]: Yes it is not used anywhere outside test code. Is this a utility class for Hazelcast users? I do not know.

File: hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/instance/HazelcastInstanceLoader.java [hunk 2]
```diff
@@ -39,63 +39,51 @@
...
     public void configure(Properties props) {
-        this.props.putAll(props);
-    }
-
-    public HazelcastInstance loadInstance() throws CacheException {
-        if (instance != null && instance.getLifecycleService().isRunning()) {
-            LOGGER.warning("Current HazelcastInstance is already loaded and running! "
-                    + "Returning current instance...");
-            return instance;
-        }
```
Comment (mesutcelik) [inline hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/instance/HazelcastInstanceLoader.java]: I see this block is not deleted in HazelcastClientLoader. Is this intentional?

Comment (mustafaiman) [inline hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/instance/HazelcastInstanceLoader.java]: should be deleted there, too. i'll fix it. thanks

File: hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/instance/HazelcastAccessor.java (hunk 1 above)
Comment (kobalski) [inline hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/instance/HazelcastAccessor.java]: I think it is designed to be a utility class to access hazelcast instance used by hibernate. See the old google groups post about it : https://groups.google.com/forum/#!searchin/hazelcast/HazelcastAccessor%7Csort:relevance%7Cspell:false/hazelcast/kVZfayIhjYA/Voh66NHOo1IJ

Comment (mesutcelik) [inline hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/instance/HazelcastAccessor.java]: so they can use `instanceName` for the same purpose?

Comment (mustafaiman) [inline hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/instance/HazelcastAccessor.java]: They can start hibernate without specifying an instanceName for Hazelcast. They need to access the instance first to learn the instanceName in that case.

Comment (mustafaiman) [inline hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/instance/HazelcastAccessor.java]: I moved HazelcastAccessor under test directory.

Comment (kobalski) [inline hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/instance/HazelcastAccessor.java]: I think we should not remove or move `HazelcastAccessor` class, some users might be using this utility class.

Comment (mesutcelik) [inline hazelcast-hibernate/hazelcast-hibernate3/src/main/java/com/hazelcast/hibernate/instance/HazelcastAccessor.java]: my vote goes to removal... `instanceName` is hazelcast way of sharing a running instance. Less code is less maintenance... @mdogan ?

**Comment by pveentjer** [comment]
Can you provide a better title/description? Because based on it, I'm not able to determine which part of the system was changed.

**Comment by mustafaiman** [comment]
@pveentjer i updated title and description. can you take a look?

**Comment by mustafaiman** [comment]
I will send another pr

**Pull request closed without merging**